File: silver/1620/c++/main.cpp
#include <iostream>
#include <unordered_map>
#include <string>

int main()
{
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);

    int n, m;
    std::cin >> n >> m;
    std::unordered_map<int, std::string> numberToString;
    std::unordered_map<std::string, int> stringToNumber;
    std::string name;

    for (int i = 1; i <= n; i++) {
        std::cin >> name;
        numberToString[i] = name;
        stringToNumber[name] = i;
    }

    std::string text;
    int num;
    for (int i = 0; i < m; i++) {
        std::cin >> text;
        if (std::isdigit(text[0])) {
            int num = std::stoi(text);
            std::cout << numberToString[num] << '\n';
        } else {
            std::cout << stringToNumber[text] << '\n';
        }
    }

    return 0;
}

File: silver/1620/python/main.py
import sys
input = sys.stdin.readline

if (__name__ == "__main__"):
    n, m = map(int, input().split())
    numberToString = dict()
    stringToNumber = dict()
    for i in range(1, n +1):
        name = input().rstrip()
        numberToString[i] = name
        stringToNumber[name] = i

    for i in range(m):
        text = input().rstrip()
        if (text.isdigit()):
            num = int(text)
            print(numberToString[num])
        else:
            print(stringToNumber[text])


